ViewLayer/CompactView/CompactView.cpp

```diff
@@ -0,0 +1,491 @@
+      ui_.batteryCMU4Cell3VoltageLabel().setText(QString::number(cellVoltages[2], 'f', 3));
+      ui_.batteryCMU4Cell4VoltageLabel().setText(QString::number(cellVoltages[3], 'f', 3));
+      ui_.batteryCMU4Cell5VoltageLabel().setText(QString::number(cellVoltages[4], 'f', 3));
+      ui_.batteryCMU4Cell6VoltageLabel().setText(QString::number(cellVoltages[5], 'f', 3));
+      ui_.batteryCMU4Cell7VoltageLabel().setText(QString::number(cellVoltages[6], 'f', 3));
+      ui_.batteryCMU4Cell8VoltageLabel().setText(QString::number(cellVoltages[7], 'f', 3));
+   }
+}
+
+void CompactView::highlightMinMaxVoltages()
+{
+   QLabel* newMaxVoltageLabel = ui_.batteryCMUCellVoltageLabels()[0];
+   double newMaxVoltage = newMaxVoltageLabel->text().toFloat();
+   QLabel* newMinVoltageLabel = ui_.batteryCMUCellVoltageLabels()[0];
+   double newMinVoltage = newMinVoltageLabel->text().toFloat();
+
+   foreach(QLabel* cursorVoltageLabel, ui_.batteryCMUCellVoltageLabels())
+   {
+      double cursorVoltage = cursorVoltageLabel->text().toFloat();
+      cursorVoltageLabel->setStyleSheet("");
+      if(cursorVoltage > 0)
+      {
+         if(cursorVoltage > newMaxVoltage)
+         {
+            newMaxVoltageLabel = cursorVoltageLabel;
+            newMaxVoltage = cursorVoltage;
+         }
+         if(cursorVoltage < newMinVoltage)
+         {
+            newMinVoltageLabel = cursorVoltageLabel;
+            newMinVoltage = cursorVoltage;
+         }
+     }
+   }
+   newMaxVoltageLabel->setStyleSheet("font-weight: bold; background-color: rgb(100, 100, 100);");
+   newMinVoltageLabel->setStyleSheet("font-weight: bold; background-color: rgb(100, 100, 100);");
+   highlightUntrustedVoltages();
+}
+
+void CompactView::highlightUntrustedVoltages()
+{
+   foreach(QLabel* cursorVoltageLabel, ui_.batteryCMUCellVoltageLabels())
+   {
+      if(cursorVoltageLabel->text().toFloat() < 0)
+      {
+         QString stylesheet = cursorVoltageLabel->styleSheet();
+         cursorVoltageLabel->setStyleSheet(stylesheet + "color: yellow");
+         cursorVoltageLabel->setText(cursorVoltageLabel->text().right(1));
+      }
+   }
+}
+
+void CompactView::connectionFailed(const QString& errorString)
+{
+   ui_.inputSerialConnectionStatusLabel().setText(errorString);
+   ui_.inputSerialConnectionStatusLabel().setStyleSheet("text-align: centre; color: rgb(255, 40, 40); background-color: rgb(70,70,70);"); // red text
+}
+
+void CompactView::connectionSucceeded()
+{
+   ui_.inputSerialConnectionStatusLabel().setText("Connected");
+   ui_.inputSerialConnectionStatusLabel().setStyleSheet("text-align: centre; color: rgb(0, 255, 0); background-color: rgb(70,70,70);");
+}
+void CompactView::motorOneFaultsReceived(MotorFaults motorFaults)
+{
+   if(motorFaults.softwareOverCurrent() && !isLabelRed(ui_.softwareOverCurrentLabel1())){
+      ui_.softwareOverCurrentLabel1().setStyleSheet("background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,stop: 0 rgb(176, 13, 13), stop: 1 red);");
```

Comment:
I think this functionality should be abstracted into another function. Turning a faultLabel red or green.